### PR TITLE
Don't normalize property casing

### LINKED ISSUE #185: Issues due to property normalizing
I'm trying to integrate against a SOAP service and I'm running into some issues with the property normalizing.

All properties in the WSDL are in UpperCamelCase (e.g. OutErrorCode, OutNumAvail etc.) and are normalized during code generation to lowerCamelCase (e.g. `$outErrorCode`, `$outNumAvail` etc.). So far so good!

However, problems occur when making requests and trying to use the result classes since our friend SoapClient seems to assign values to the properties' UpperCamelCase variant.

Dumping the result (using `symfony/var-dumper`) outputs something along the lines of:

```
FoobarResponse {#101
  -outErrorCode: null
  -outFoobar: null
  +"OutErrorCode": "11"
  +"OutFoobar": ""
}
```

I assume this isn't the intended behaviour? The workaround right now is to manually change all type properties to the correct case. My assumption is that SoapClient is unaware of the `OutErrorCode -> outErrorCode` transformation, but I'm not sure it's possible to make it aware either.

Thanks!

### PR DESCRIPTION
**Pull request by janvernieuwe**
Fixes #185

### CODE REVIEW

File: spec/Phpro/SoapClient/CodeGenerator/Util/NormalizerSpec.php [hunk 1]
```diff
@@ -30,14 +30,21 @@ function it_can_normalize_classnames()
...
+        $this->normalizeClassname('UpperCased')->shouldReturn('UpperCased');
         $this->normalizeClassname('my-./*type_123')->shouldReturn('MyType123');
+        $this->normalizeClassname('my-./final*type_123')->shouldReturn('MyFinalType123');
     }
 
     function it_noramizes_properties()
     {
         $this->normalizeProperty('prop1')->shouldReturn('prop1');
-        $this->normalizeProperty('final')->shouldReturn('finalType');
+        $this->normalizeProperty('final')->shouldReturn('final');
```
Comment (veewee) [inline spec/Phpro/SoapClient/CodeGenerator/Util/NormalizerSpec.php]: This is perfectly valid php:
```php
class X {
    private $final;
    private $global;
    private $if;
}
```
https://3v4l.org/GW5aN

If you transform those keyworded properties, ext-soap won't be able to link the xml to the objects.

File: src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php [hunk 2]
```diff
@@ -110,10 +110,9 @@ private static function normalizeReservedKeywords(string $name, $ucfirst = true)
         if (!\in_array(strtolower($name), self::$reservedKeywords, true)) {
             return $name;
         }
-        $name = ucfirst($name);
         $name .= 'Type';
 
-        return $ucfirst ? ucfirst($name) : lcfirst($name);
+        return $ucfirst ? ucfirst($name) : $name;
```
Comment (veewee) [inline src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php]: Maybe better to drop the ucfirst param and wrap the function where you need it? Seems easier to follow.

Comment (janvernieuwe) [reply src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php]: Agreed, was thinking about it yesterday.

**Comment by janvernieuwe** [comment]
Maybe this can fix #186 too, but there is not enough information in that issue to be sure of that.

**Pull request merged**